### PR TITLE
Loadout optimizer optimizations

### PR DESCRIPTION
**Pull request by bhollis**
This should drastically reduce the occurrences of sets needing to recalculate (including embarassingly twice upon fresh load), and speeds up the actual processing by reducing garbage created during the loops (~20% speedup).

### CODE REVIEW

File: src/app/loadout-builder/process.ts [hunk 1]
```diff
@@ -250,20 +246,23 @@ export function process(
...
+                ];
+
                 const maxPower = getPower(firstValidSet);
+
+                const stats = {};
                 for (const stat of statChoices) {
                   let index = 0;
                   for (const key of statKeys) {
-                    stats[key] += stat[index];
+                    stats[key] = (stat[key] || 0) + stat[index];
```
Comment (ryan-rushton) [inline src/app/loadout-builder/process.ts]: I think this should be `stats[key] = (stats[key] || 0) + stat[index];`, it creating a few problems atm.

**Comment by ryan-rushton** [comment]
I am seeing this double run of process happening just before the load stores timed log is put in the console. It's also happening in master so I figure I would just bring it up because we are on the topic of performance.
image

File: src/app/loadout-builder/process.ts (hunk 1 above)
Comment (bhollis) [reply src/app/loadout-builder/process.ts]: You're right.

**Comment by bhollis** [comment]
I think that's what this branch is fixing? Or are you saying you see that within this PR?

**Comment by bhollis** [comment]
Ah, looks like some of the intervening changes broke some of the optimizations. I'll dig in again.

**Pull request merged**